hs_collection_resource/tests/test_collection.py

```diff
@@ -527,3 +536,83 @@ def test_versioning(self):
+
+        # update resGeoFeature coverage
+        metadata_dict = [{'coverage': {'type': 'point', 'value': {'name': 'Name for point coverage', 'east': '-20',
+                                                     'north': '10', 'units': 'decimal deg'}}},]
+        update_science_metadata(pk=self.resGeoFeature.short_id, metadata=metadata_dict)
+        self.assertEqual(self.resGeoFeature.metadata.coverages.count(), 1)
+        # calculate overall coverages
+        _update_collection_coverages(self.resCollection)
+
+        # collection should have 2 coverage metadata: period and point
+        self.assertEqual(self.resCollection.metadata.coverages.count(), 2)
+        found_period = False
+        found_point = False
+        for cv in self.resCollection.metadata.coverages.all():
+            if cv.type == 'period':
+                found_period = True
+                self.assertEqual(parser.parse(cv.value['start'].lower()), parser.parse('1/1/2016'))
+                self.assertEqual(parser.parse(cv.value['end'].lower()), parser.parse('12/31/2016'))
+            if cv.type == 'point':
+                found_point = True
+                self.assertEqual(cv.value['east'], -20)
+                self.assertEqual(cv.value['north'], 10)
+        self.assertTrue(found_period)
+        self.assertTrue(found_point)
+
+        # add a 3rd res with period and box coverages into collection
+        metadata_dict = [{'coverage': {'type': 'period', 'value': {'name': 'Name for period coverage', 'start': '1/1/2010','end': '6/1/2016'}}},
+                         {'coverage': {'type': 'point', 'value': {'name': 'Name for point coverage', 'east': '25',
+                                                     'north': '-35', 'units': 'decimal deg'}}}
+                        ]
+        update_science_metadata(pk=self.resGen2.short_id, metadata=metadata_dict)
+        self.assertEqual(self.resGen2.metadata.coverages.count(), 2)
+        self.resCollection.resources.add(self.resGen2)
+        self.assertEqual(self.resCollection.resources.count(), 3)
+        # calculate overall coverages
+        _update_collection_coverages(self.resCollection)
+        found_period = False
+        found_box = False
+        self.assertEqual(self.resCollection.metadata.coverages.count(), 2)
+        for cv in self.resCollection.metadata.coverages.all():
+            if cv.type == 'period':
+                found_period = True
+                self.assertEqual(parser.parse(cv.value['start'].lower()), parser.parse('1/1/2010'))
+                self.assertEqual(parser.parse(cv.value['end'].lower()), parser.parse('12/31/2016'))
+            if cv.type == 'box':
+                found_box = True
+                self.assertEqual(cv.value['westlimit'], -20)
+                self.assertEqual(cv.value['northlimit'], 10)
+                self.assertEqual(cv.value['eastlimit'], 25)
+                self.assertEqual(cv.value['southlimit'], -35)
+        self.assertTrue(found_period)
```

Comment:
You can do this instead:
``self.assertTrue(self.resCollection.metadata.coverages.all().filter(type='period').exists())``